geomstats/learning/frechet_mean.py

```diff
@@ -72,59 +67,55 @@ def linear_mean(points, weights=None):
     mean : array-like, shape=[1, dimension]
         Weighted linear mean of the points.
     """
+    # TODO(ninamiolane): Factorize this code to handle lists
+    # in the whole codebase
     if isinstance(points, list):
-        points = gs.vstack(points)
-    points = gs.to_ndarray(points, to_ndim=2)
-    n_points = gs.shape(points)[0]
-
+        points = gs.stack(points, axis=0)
     if isinstance(weights, list):
-        weights = gs.vstack(weights)
-    elif weights is None:
-        weights = gs.ones((n_points, ))
+        weights = gs.stack(weights, axis=0)
+
+    n_points = geomstats.vectorization.get_n_points(
+        points, point_type)
+
+    if weights is None:
+        weights = gs.ones((n_points,))
 
     weighted_points = gs.einsum('...,...j->...j', weights, points)
-    mean = (gs.sum(weighted_points, axis=0) / gs.sum(weights))
-    mean = gs.to_ndarray(mean, to_ndim=2)
+    mean = gs.sum(weighted_points, axis=0) / gs.sum(weights)
```

Comment:
I don't think L83 works if points is 3d (e.g. matrix point_type?)

Reply:
Fixed.